simulation/src/transportation/bird_mob.rs

```diff
@@ -0,0 +1,137 @@
+use crate::physics::{Collider, CollisionWorld, PhysicsGroup, PhysicsObject, Speed};
+use crate::utils::rand_provider::RandProvider;
+use crate::utils::resources::Resources;
+use crate::utils::time::GameTime;
+use crate::World;
+use common::rand::rand3;
+use egui_inspect::Inspect;
+use geom::{angle_lerpxy, Transform, Vec3};
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Inspect)]
+pub struct BirdMob {
+    pub walking_speed: f32,
+    pub fly_anim: f32,
+}
+
+const PED_SIZE: f32 = 0.5;
+
+pub fn put_bird_in_coworld(coworld: &mut CollisionWorld, pos: Vec3) -> Collider {
+    Collider(coworld.insert(
+        pos.xy(),
+        PhysicsObject {
+            radius: PED_SIZE * 0.6,
+            group: PhysicsGroup::Unknown,
+            ..Default::default()
+        },
+    ))
+}
+
+impl BirdMob {
+    pub(crate) fn new(r: &mut RandProvider) -> Self {
+        Self {
+            walking_speed: (10.0 + r.next_f32() * 0.4),
+            fly_anim: 0.0,
+        }
+    }
+}
+
+pub fn bird_decision_system(world: &mut World, resources: &mut Resources) {
+    profiling::scope!("transportation::animal_decision_system");
+    let ra = &*resources.read();
+    let map = &*resources.read::<Map>();
+    let next_dests: Vec<Vec3> = map
+        .buildings()
+        .values()
+        .map(|building| building.door_pos.up(-building.height))
+        // .chain(
+        //     map.terrain
+        //         .trees
+        //         .objects()
+        //         .map(|tree| vec3(tree.1.pos.x, tree.1.pos.y, tree.1.size)),
+        // )
+        .collect();
+    world.birds
+        .values_mut()
+        //.par_bridge()
+        .for_each(|human| bird_decision(ra, &mut human.it, &mut human.trans, &mut human.speed, &mut human.bird_mob, &next_dests))
+}
+
+const BIRD_WAIT_TIME: f64 = 100.0;
+
+pub fn bird_decision(
+    time: &GameTime,
+    it: &mut Itinerary,
+    trans: &mut Transform,
+    kin: &mut Speed,
+    bird_mob: &mut BirdMob,
+    next_dests: &Vec<Vec3>,
+) {
+    let (desired_v, desired_dir) = calc_decision(bird_mob, trans, it);
+
+    bird_mob.fly_anim += 2.0 * kin.0 * time.realdelta / bird_mob.walking_speed;
+    bird_mob.fly_anim %= 2.0 * std::f32::consts::PI;
+    physics(kin, trans, time, desired_v, desired_dir);
+
+    let get_new_itinerary = || {
+        let n2 = rand3(trans.position.y, trans.position.x, time.timestamp as f32);
+        let n1 = rand3(trans.position.x, trans.position.y, time.timestamp as f32);
+        if n1 > 0.5 {
+            Itinerary::simple(vec![next_dests[(next_dests.len() as f32 * n2) as usize]])
+        } else {
+            Itinerary::wait_until(time.timestamp + BIRD_WAIT_TIME)
+        }
+    };
+
+    // choose a random new destination if the current one has been reached
+    match it.end_pos() {
+        Some(end_pos) => {
+            if trans.position.is_close(end_pos, 3.0) {
+                *it = get_new_itinerary();
+            }
+        }
+        None => {
```

Comment:
you don't need to check the end_pos, has_ended is enough as a simple itinerary will automatically detect when the bird has arrived to its destination.